app/trip.py

```diff
@@ -0,0 +1,75 @@
+import datetime
+from math import dist
+from app.shop import Shop
+from app.customer import Customer
+
+
+def calculate_cost_of_the_trip(
+    customer: Customer, fuel_price: float, shop: Shop
+) -> float:
+    distance = dist(customer.location, shop.location)
+    cost_1 = round(
+        (customer.car.fuel_consumption / 100)
+        * (distance * 2) * fuel_price, 2
+    )
+    cost_2 = 0
+    for product, quantity in customer.wanted_products.items():
+        cost_2 += shop.provided_products.get(product, 0) * quantity
+    return cost_1 + cost_2
+
+
+def choose_shop(customer: Customer,
+                shops: list[Shop],
+                fuel_price: float) -> int:
+    result = []
+    for shop in shops:
+        trip_costs = calculate_cost_of_the_trip(
+            customer, fuel_price, shop
+        )
+        result.append(trip_costs)
+        print(f"{customer.name}'s trip to the "
+              f"{shop.name} " f"costs {trip_costs}")
+    return result.index(min(result))
```

Comment:
You can choose the min price directly in `for shop in shops` loop
No need to make redundant iterations
And you can return both the shop instance and the trip_costs